source/D2Common/src/D2Inventory.cpp

```diff
@@ -3342,19 +3342,13 @@ BOOL __stdcall INVENTORY_RemoveAllItems(D2InventoryStrc* pInventory)
 
 	pInventory->pCursorItem = nullptr;
 	pInventory->dwLeftItemGUID = -1;
-
-	D2UnitStrc* pItem = pInventory->pFirstItem;
-	if (pItem)
+	
+	for (D2UnitStrc* pItem = pInventory->pFirstItem; pItem; pItem = pInventory->pFirstItem)
```

Comment:
Maybe a while loop such as this is even better ? (no need for a variable)

```cpp
while(pInventory->pFirstItem != nullptr)
{
    if(!INVENTORY_RemoveItem(pInventory->pFirstItem))
    {
        return FALSE;
    }
}
```

Reply:
Yeah, that's even better